src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIProjectsService.java

```diff
@@ -49,14 +50,37 @@
 	private final List<String> PROJECT_ROLES = ImmutableList.of("PROJECT_USER", "PROJECT_OWNER");
 
 	@Autowired
-	public UIProjectsService(ProjectService projectService, SampleService sampleService, MessageSource messageSource, ProjectOwnerPermission projectOwnerPermission, ManageLocalProjectSettingsPermission projectMembersPermission) {
+	public UIProjectsService(ProjectService projectService, SampleService sampleService, MessageSource messageSource,
+			ProjectOwnerPermission projectOwnerPermission,
+			ManageLocalProjectSettingsPermission projectMembersPermission) {
 		this.projectService = projectService;
 		this.sampleService = sampleService;
 		this.messageSource = messageSource;
 		this.projectOwnerPermission = projectOwnerPermission;
 		this.projectMembersPermission = projectMembersPermission;
 	}
 
+	/**
+	 * Create a new project based on a request sent by the UI.
+	 *
+	 * @param request {@link CreateProjectRequest} containing (at the very least) the name for the project
+	 * @return the identifier for the newly created project
+	 */
+	public Long createProject(CreateProjectRequest request) {
```

Comment:
This method should throw an exception if the project can't be read. In ProjectsAjaxController.java we catch the error and return an error message to the UI if a project is not successfully created

Reply:
Updated in ba7f02587c5e040b88053bc6ad36d99cd0e1ce66